message/message.go

```diff
@@ -0,0 +1,105 @@
+package message
+
+import "strings"
+
+// Message represents DNS messages
+type Message struct {
+	header  *header
+	queries []*query
+	Answers []*Record
+}
+
+type header struct {
+	ID      uint16
+	QR      byte
```

Comment:
Whichever caller that uses headers will need to know how to handle bytes. Is there a reason why we don't use  integers?